compiler/cpp/src/generate/t_swift_3_generator.cc

```diff
@@ -0,0 +1,2458 @@
+    out << indent() << "if let error = error";
+    block_open(out);
+    out << indent() << error_completion_call << endl;
+    block_close(out);
+
+    if (!is_oneway) {
+      out << indent() << "do";
+      block_open(out);
+      indent(out);
+      if (!ret_is_void) {
+        out << "let result = ";
+      }
+      out << "try self.recv_" << (*f_iter)->get_name() << "(on: proto)" << endl;
+
+      out << indent() << (ret_is_void ? "completion(nil)" : "completion(result, nil)") << endl;
+      block_close(out);
+
+      out << indent() << "catch let error";
+      block_open(out);
+      out << indent() << error_completion_call << endl;
+
+      block_close(out);
+    } else {
+      out << indent() << "completion(nil)" << endl;
+    }
+
+    block_close(out);
+
+    block_close(out);
+
+  }
+
+  block_close(out);
+
+  out << endl;
+}
+
+/**
+ * Generates a service server implementation.
+ *
+ * Implemented by generating a block for each service function that
+ * handles the processing of that function. The blocks are stored in
+ * a map and looked up via function/message name.
+ *
+ * @param tservice The service to generate an implementation for
+ */
+void t_swift_3_generator::generate_swift_service_server_implementation(ofstream& out,
+                                                                     t_service* tservice) {
+
+  string name = tservice->get_name() + "Processor";
+
+  indent(out) << "extension " << name << " : TProcessor";
+  block_open(out);
+
+  out << endl;
+
+  indent(out) << "static let processorHandlers: ProcessorHandlerDictionary =";
+  block_open(out);
+
+  out << endl;
+
+  out << indent() << "var processorHandlers = ProcessorHandlerDictionary()" << endl << endl;
+
+  // generate method map for routing incoming calls
+  vector<t_function*> functions = tservice->get_functions();
+  vector<t_function*>::const_iterator f_iter;
+  for (f_iter = functions.begin(); f_iter != functions.end(); ++f_iter) {
+
+    t_function* tfunction = *f_iter;
+
+    string args_type = function_args_helper_struct_type(tservice, *f_iter);
+
+    out << indent() << "processorHandlers[\"" << tfunction->get_name() << "\"] = { sequenceID, inProtocol, outProtocol, handler in" << endl
+        << endl;
+
+    indent_up();
+    out << indent() << "let args = try " << args_type << ".read(from: inProtocol)" << endl
+        << endl
+        << indent() << "try inProtocol.readMessageEnd()" << endl
+        << endl;
+
+    if (!tfunction->is_oneway() ) {
+      string result_type = function_result_helper_struct_type(tservice, tfunction);
+      indent(out) << "var result = " << result_type << "()" << endl;
+
+      indent(out) << "do";
+      block_open(out);
+
+      indent(out);
+      if (!tfunction->get_returntype()->is_void()) {
+        out << "result.success = ";
+      }
+      out << "try handler." << tfunction->get_name() << "(";
+
+      t_struct* arg_struct = tfunction->get_arglist();
+      const vector<t_field*>& fields = arg
```

Comment:
Due to the way thrift serializes method responses, using a Generic `Response<T, Error>` type pattern isn't doable since there can be multiple exception types that each have unique field ID's.  I visited this idea in development and decided against it.
Alternatively, I'm thinking of using Enum's as unions here (for both async and regular clients) to remedy unnecessary optionals here, so that the client can handle possible cases in a switch.